api/src/main/java/org/apache/cloudstack/api/ApiConstants.java

```diff
@@ -472,11 +473,16 @@
     public static final String PROJECT = "project";
     public static final String ROLE = "role";
     public static final String ROLE_ID = "roleid";
+    public static final String ACCOUNT_ROLE_TYPE = "accountroletype";
+    public static final String PROJECT_ROLE_ID = "projectroleid";
+    public static final String PROJECT_ROLE_NAME = "projectrolename";
+    public static final String PROJECT_ROLE_TYPE = "projectroletype";
     public static final String ROLE_TYPE = "roletype";
     public static final String ROLE_NAME = "rolename";
     public static final String PERMISSION = "permission";
     public static final String RULE = "rule";
     public static final String RULE_ID = "ruleid";
+    public static final String PROJECT_RULE_ID = "projectruleid";
```

Comment:
nit - probably this means projectrolepermissionid? (yes, naming is hard, I'm trying to guess what is parameter is for)